### ISSUE TITLE
Bus error on 20 characters or more.

### ISSUE THREAD

**Issue by phaer** [kallaballa/Janosh#14]
I am getting a bus error, without a stack strace or something else while trying to send strings which are 20 characters or longer. The bug is  reproduceable by running the following minimal lua script...
~~~~
#!/lounge/bin/janosh -f

function receive(handle, message)
   print(message)
end

Janosh:wsOpen(8080)
Janosh:wsOnReceive(receive)

while true do
   Janosh:sleep(1000)
end
~~~~
...and pasting the following snippet in your browsers javascript console.
~~~~
var socket = new WebSocket('ws://10.20.30.40:8080');
socket.onopen = function() { socket.send('a'.repeat(20)); }
~~~~
...which kills the server process. It works if one sends less than 20 characters.

Any ideas?

**Comment by kallaballa** [comment]
interesting. that bug is only triggered on the screeninvader. on opensuse 13.2 (x86_64) it works. i'm investigating

**Comment by kallaballa** [comment]
there was a race condition in c++ connection handles to lua connection handles mapping of the websocket implementation. fixed: 1d1b8b0a3d177a5490296699c76c89836b359463

**Comment by kallaballa** [comment]
additionally we encounter this bug: https://github.com/zaphoyd/websocketpp/issues/395

**Comment by kallaballa** [comment]
fixed: d890c01e8d4798a9221e1ae5d9886ed439f5cdbd
